src/main/java/org/terasology/climateConditions/ModifyThirstSystem.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.climateConditions;
+
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.BeforeRemoveComponent;
+import org.terasology.entitySystem.entity.lifecycleEvents.OnAddedComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.thirst.component.ThirstComponent;
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class ModifyThirstSystem extends BaseComponentSystem {
+    private float defaultNormalDecayPerSecond;
+    private float defaultSprintDecayPerSecond;
+
+    @ReceiveEvent(components = {HyperthermiaComponent.class})
+    public void onAddedHypothermia(OnAddedComponent event, EntityRef player, ThirstComponent thirst) {
+        scaleThrist(player, thirst, 2f);
+    }
+
+    @ReceiveEvent(components = {HyperthermiaComponent.class})
+    public void beforeRemoveHyperthermia(BeforeRemoveComponent event, EntityRef player, ThirstComponent thirst) {
+        getDefaultThirst(player);
+        thirst.normalDecayPerSecond = defaultNormalDecayPerSecond;
+        thirst.sprintDecayPerSecond = defaultSprintDecayPerSecond;
+        player.saveComponent(thirst);
+    }
+
+
+    public void scaleThrist(EntityRef player, ThirstComponent thirst, float magnitude) {
+        thirst.normalDecayPerSecond *= magnitude;
+        thirst.sprintDecayPerSecond *= magnitude;
+        player.saveComponent(thirst);
+    }
+
+    public void getDefaultThirst(EntityRef player) {
+        ThirstComponent thirst = player.getParentPrefab().getComponent(ThirstComponent.class);
+        defaultNormalDecayPerSecond = thirst.normalDecayPerSecond;
+        defaultSprintDecayPerSecond = thirst.sprintDecayPerSecond;
```

Comment:
I believe that this will not give you the default (original) thirst decay, but the one you set when adding `HyperthermiaComponent` :thinking:

Reply:
I'm sorry, Do you want beforeRemoveHyperthermia to change value to the one before adding Hyperthermia, or do you think this code will give the value of thirst decay before adding Hyperthermia and want it to actually change it to the default value?
Although I'm pretty sure it's the first one, so I did divide it by magnitude at first but at the time I was scaling it by 2.25 (or some decimal) and then the thing with floating point no.s about multiplying and dividing with the same no. does not result in the original value again got to me.

Reply:
Maybe I misunderstood the intention behind the code... Let me ask you this: What information will be stored (when) in `defaultNormalDecayPerSecond` and `defaultSprintDecayPerSecond`?

Reply:
Ohhkay... my intention was to retrieve the default values of the two decays using the parent prefab (I thought that the parent prefab has the default values irrespective of the values in the component the player entity possesses at the time of being retrieved....) So  the intention as such was to retrieve these "default" values right before exiting the state of hyperthermia.

Reply:
Oohhh... totally overlooked the `getParentPrefab` :facepalm:
my ability to focus is not so good today it seems :roll_eyes: 
sorry 'bout that and thanks for the explanations and your patience :sweat_smile: